data/virt_autotest/guest_console_monitor.sh

```diff
@@ -0,0 +1,240 @@
+      help_usage
+      exit 1
+      ;;
+    *)
+      help_usage
+      exit 1
+      ;;
+  esac
+done
+
+#FUNCTION PART
+function get_console()
+{
+
+local vmguest=$1
+
+expect -c "
+set timeout 60
+
+##hide echo
+log_user 0
+spawn -noecho virsh console ${vmguest}
+
+#wait connection
+sleep 3
+send \"\r\n\r\n\r\n\"
+
+#condition expect
+expect {
+        \"*login:\" {
+                send \"root\r\"
+                exp_continue
+        }
+        -nocase "password:" {
+                send \"novell\r\"
+                exp_continue
+        }
+        \"*:~ #\" {
+                send -- \"ip route get 1\r\"
+        }
+        timeout {
+                send -- \"exit\r\"
+                exp_continue        
+        }                 
+}
+
+## -1 means never timeout
+set timeout -1
+
+expect -re {dev.*\s([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})[^0-9]}
+
+if {![info exists expect_out(1,string)]} {
+        puts \"Match did not happen :(\"
+        exit 1
+}
+set output \$expect_out(buffer)
+expect eof
+
+puts \"\$output\"
+"
+
+}
+
+
+function open_console()
+{
+        #open guest console
+        for vmguest in ${vm_guestnames_state_running_array[@]};do
+                get_console ${vmguest} >> ${LOG_DIR}/${vmguest}_console_${timestamp} 2>&1 &
+        done 
+}
+
+function close_console()
+{
+        #close guest console
+        echo "kill PIDs $(basename $0)"
+        pid=`ps aux | grep "$(basename $0) -s" | grep -v "grep"| awk '{print $2}'`
+        if [[ -n $pid ]];then
+            kill -9 $pid >> /dev/null 2>&1;sync
+        else
+            echo "PIDs $(basename $0) were killed"
+        fi        
+        
+        echo "Kill all associated "virsh console" processes"
+        ps aux | grep "virsh console" | grep -v "grep" >> /dev/null 2>&1
+        while [[ $? -eq 0 ]];    
+        do
+            #wait for checking all associated "virsh console" processes"
+            sleep 3
+            sub_pid=`ps aux | grep "expect -c" | grep -v "grep"| awk '{print $2}'`
+            if [[ -n $sub_pid ]];then 
+                kill -9 $sub_pid >> /dev/null 2>&1;sync
+            else
+                break
+            fi    
+        done        
+}
+
+
+
+function end_monitor()
+{
+        #end of monitor for guest console
+	if [ -f ${MONITOR_LOCKED_FILE} ];then
+            echo "End of Monitor for guest console"	
+            #UNLOCK Monitor for guest console	    
+            > ${MONITOR_LOCKED_FILE} ; rm -rf ${MONITOR_LOCKED_FILE}
+            #Kill Pids from Monitor for guest console
+            close_console
+        fi
+}
+
+function start_monitor()
+{
+        #start monitor for guest console
+        get_vm_guestnames_state_running=`virsh list --state-running | grep "${vm_guestnames_types}" | awk '{print $2}'`
+        vm_guestnames_state_running_array=$(echo -e ${get_vm_guestnames_state_running})  
+
+        while :
+        do
+		if [ -f ${MONITOR_LOCKED_FILE} ];then
+                #Check with LOCK-UP status
```

Comment:
I am sorry. I do not get why this lock file is introduced and how it helps. Can you help explain?

Reply:
As mentioned, Start Monitor for guest console
Collect the existed vm guest(s) on vm host -> try to boot up the existed vm guest(s).
-> **Create LOCK-UP file to save PID(s) from virsh console as background**
-> call get_console() to made virsh console as child process, which was ran as background PID and redirect
output of virsh console to specified log file

instance 1:
Trigger this script to start Monitor for guest console, then try to this script to start Monitor for guest console again as
twice, this LOCK-UP file will save the existed virsh console PID